scripts/concierge.ts

```diff
@@ -0,0 +1,164 @@
+/* eslint-disable no-console */
+import hre from 'hardhat'
+import fs from 'fs'
+
+import {OwnedNft, OwnedNftsResponse} from 'alchemy-sdk'
+import {
+    batchFetchClaimed,
+    getDelegateAddresses,
+    loadCSV,
+    requestOwnerNfts,
+    requestProofs
+} from './concierge_helper'
+
+const PROJECT_ID = process.env.PROJECT_ID || ''
+
+interface ClaimData {
+    recipient: string
+    proof: string[]
+    group: string
+    data: string
+    leaf: string
+    amount: string
+    unlockingAt: number
+    expiryTimetamp: number
+    index: number
+    expiryTimestamp: number
+}
+
+export interface IAirdropClaim {
+    claimable?: boolean
+    recipient: string
+    index: number
+    proof: string[]
+    amount: string
+    unlockingAt: number
+    group: string
+    data: string
+    leaf: string
+    claimed?: boolean
+    expiryTimestamp: number
+    expired?: boolean
+    nft?: OwnedNft
+}
+
+function chunk<T>(array: T[], chunkSize: number): T[][] {
+    const chunkedArray: T[][] = []
+    let index = 0
+
+    while (index < array.length) {
+        chunkedArray.push(array.slice(index, index + chunkSize))
+        index += chunkSize
+    }
+
+    return chunkedArray
+}
+
+async function run() {
+    const delegatedAccountPk = process.env.CONCIERGE_PRIVATE_KEY || ''
+    const distributorAddress = process.env.DISTRIBUTOR_CONTRACT || ''
+
+    const signer = new hre.ethers.Wallet(
+        delegatedAccountPk,
+        hre.ethers.provider
+    )
+
+    const delegatingAddresses = await getDelegateAddresses(
+        signer.address as `0x${string}`
+    )
+
+    const executeTime = Date.now()
+    console.log(
+        `Script started for NFT airdrop claiming on behalf: project ${PROJECT_ID}, contract ${distributorAddress} @ ${executeTime}`
+    )
+
+    const csvPath = 'concierge_address.csv'
+    const csvUsers = await loadCSV(csvPath)
+
+    console.log(csvUsers)
+
+    const userAddresses =
+        delegatingAddresses?.filter((add) => csvUsers.includes(add)) || []
```

Comment:
should we normalize the casing?
```suggestion
        delegatingAddresses?.filter((add) => csvUsers.includes(add.toLowerCase())) || []
```